[DOCS] Updated contribution docs (#7880)
great-expectations/great_expectations: CONTRIBUTING_CODE.md

pip install -e .
```

`-e` will install Great Expectations in “editable” mode. This is not required, but is often very convenient as a developer.

> 

Reviewer: 🚀

- Java. See [Java downloads](https://www.java.com/en/download/).

- The PATH or JAVA_HOME environment variables set correctly. See [Setting Java variables in Windows](https://www.ibm.com/docs/en/b2b-integrator/5.2?topic=installation-setting-java-variables-in-windows) or [Setting Java variables in Linux](https://www.ibm.com/docs/en/b2b-integrator/5.2?topic=installation-setting-java-variables-in-linux). 

> On Mac, if you installed Java via the download above, you can set PATH and JAVA_HOME using:

Reviewer 2: ```suggestion
On Mac, run the following commands to set the PATH and JAVA_HOME environment variables:
```
![image](https://github.com/great-expectations/great_expectations/assets/104205257/3377434e-8b30-4b6c-9ab2-08b52188e76d)

```
export JAVA_HOME=`/usr/libexec/java_home`
export PATH=$PATH:$JAVA_HOME/bin 
```

#### Install PySpark

> - Installing PySpark will now install Spark for you. See [Spark Overview](https://spark.apache.org/docs/latest/index.html#downloading).

Reviewer 2: ```suggestion
When you install PySpark, Spark is also installed. See [Spark Overview](https://spark.apache.org/docs/latest/index.html#downloading).
```

> Run the following command to install the PySpark & Apache Spark:

Reviewer 2: ```suggestion
Run the following command to install PySpark and Apache Spark:
```
![image](https://github.com/great-expectations/great_expectations/assets/104205257/fc0df9c2-f2a4-47c4-bcd9-f6dfed926e1b)

```console
pip install pyspark
```

> ## If you want to build documentation locally:

Reviewer 2: ```suggestion
## Build documentation locally
```

> `pip install -r docs/requirements.txt`

Reviewer 2: ```suggestion
Run the following command to review the documentation requirements text file:

```console
pip install -r docs/requirements.txt
```

```

> `pip install -r docs/requirements.txt`

Reviewer 3: I think these instructions are for the legacy docs (in `docs_rtd`). We should probably replace these instructions with instructions on how to build our current docs. I'm hoping to find a bit of time today to simplify that process using an invoke command - let's coordinate.

Author: Ah sounds good. Looks like the docs here already link to the README within docs folder. I will remove this section that is no longer required.

Reviewer 3: Awesome! For reference, here is the open PR with a new method to build docs locally and updated instructions: https://github.com/great-expectations/great_expectations/pull/7892

> To build documentation, the command is `cd docs; make html`

Reviewer 2: ```suggestion
Run the follwoing command to build the documentation:

```console
cd docs; make html
```
```

Documentation will be generated in `docs/build/html/` with the index.html as the index page.

> Note: we use autoapi to generate API reference docs, but it’s not compatible with pandas 1.1.0. You’ll need to have pandas 1.0.5 (or a previous version) installed in order to successfully build docs.

Reviewer 2: ```suggestion
AutoAPI is used to generate API reference documentation, but it’s not compatible with pandas 1.1.0. To  build the API documentation, install pandas 1.0.5 or earlier.
```
Move this sentence so it appears immediately after the heading.
